### DISCUSSION TITLE
Show an indicator that a split is zoomed in

### DISCUSSION THREAD

**Discussion by zx8** [ghostty-org/ghostty#3288] [Feature Requests, Ideas]
When I'm zoomed in to a split and e.g. have multiple other tabs open, it's easy to forget that I zoomed in to a split in one tab.

Would it be possible to add an indicator that a split is zoomed in, similar to the Secure Keyboard Entry icon?

For reference, iTerm uses the following indicator:

<img width="80" alt="image" src="https://github.com/user-attachments/assets/b3b2c5dd-0756-4d58-8498-67c2170a7cfb" />

**Comment by gpanders** [comment]
There is an icon when a split is "zoomed". If there are no tabs, the icon is in the title bar:

<img width="54" alt="Screenshot 2024-12-27 at 9 37 38 AM" src="https://github.com/user-attachments/assets/9b85ab36-d697-4d96-8ab8-39c4a8af6f74" />


If there is a tab, then the icon is in the tab itself:

<img width="912" alt="Screenshot 2024-12-27 at 9 37 48 AM" src="https://github.com/user-attachments/assets/48f39c68-54ce-4470-96c7-aadff851cdf4" />

**Reply by zx8** [reply]
That's strange, I can see the icon when I have multiple tabs open, but I can't see it when I have no tabs open.

It seems to be related to the following setting:

```
macos-titlebar-style = tabs
```

When I comment it out, I can see the icon you're referring to.

**Reply by mitchellh** [reply]
Yeah this feature only applies to native tabs currently.

**Reply by ralgozino** [reply]
Also, if you have just one tab (none?) you don't see the icon when zooming in a split. I presume because the icon is drawn in the tab itself.

I just wanted to leave this documented somewhere because I was not seeing the icon with a split zoomed in and I was sure this existed and searched for a while until I realised that I was not seeing the icon anymore because I had no (other) tabs open.

**Reply by danielwe** [reply]
The `macos-titlebar-style = tabs` limitation is fixed by #7496 if accepted

**Comment by pluiedev** [comment]
Gonna keep this open since I think this is missing from Linux/GTK

**Reply by 00-kat** [reply]
Yep it indeed is.

**Reply by rhodes-b** [reply]
🔍  shows in the titlebar and tab when you have a split zoomed in GTK. This is not missing

**Reply by pluiedev** [reply]
![image](https://github.com/user-attachments/assets/36b55c80-f692-461e-a5d7-09f348bd8383)

It works in the titlebar but not in the tab for me, but I *think* I know what's happening here: I don't have color emojis installed. Truthfully though, this should instead be implemented by setting either [`AdwTabPage.set_indicator_icon`](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/method.TabPage.set_indicator_icon.html) or [`AdwTabPage.set_icon`](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/method.TabPage.set_icon.html), since the GNOME design language prescribes using symbolic (monochrome) icons in these cases.

**Reply by jalberto** [reply]
What happens when there is no tab or title bar?

What about an option to add a color border or similar?
